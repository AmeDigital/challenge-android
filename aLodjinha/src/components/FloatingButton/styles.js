import { StyleSheet } from 'react-native';

import material from '../../utils/MaterialDesign';

export default StyleSheet.create({
	TouchableOpacityStyle: {
		position: 'absolute',
		width: 50,
		height: 50,
		alignItems: 'center',
		justifyContent: 'center',
		right: 18,
		bottom: 18,
		backgroundColor: material.colorWarmPurple,
		padding: 30,
		borderRadius: 50,
		elevation: 4
	},

	FloatingButtonStyle: {
		resizeMode: 'contain'
	}
});
