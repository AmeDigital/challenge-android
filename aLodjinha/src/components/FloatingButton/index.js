import React from 'react';
import { TouchableOpacity, Image, ActivityIndicator } from 'react-native';

import styles from './styles';

export default function FloatingButton({ onPress, loading }) {
	return (
		<TouchableOpacity
			disabled={loading}
			activeOpacity={0.7}
			onPress={() => onPress()}
			style={styles.TouchableOpacityStyle}
		>
			{loading ? (
				<ActivityIndicator size="small" color="#fff" style={styles.FloatingButtonStyle} />
			) : (
				<Image source={require('../../images/check_button.png')} style={styles.FloatingButtonStyle} />
			)}
		</TouchableOpacity>
	);
}
