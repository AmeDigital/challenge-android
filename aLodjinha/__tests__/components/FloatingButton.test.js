import React from 'react';

import renderer from 'react-test-renderer';

import FloatingButton from '../../src/components/FloatingButton';

describe('Component FloatingButton', () => {
	test('should render without erros', () => {
		const tree = renderer.create(<FloatingButton />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
