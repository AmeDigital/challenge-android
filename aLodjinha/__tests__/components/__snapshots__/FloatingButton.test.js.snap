// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FloatingButton should render without erros 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5e2a84",
      "borderRadius": 50,
      "bottom": 18,
      "elevation": 4,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 30,
      "position": "absolute",
      "right": 18,
      "width": 50,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/check_button.png",
      }
    }
    style={
      Object {
        "resizeMode": "contain",
      }
    }
  />
</View>
`;
